### PR TITLE
feat(retry): implement backoff retry for Railway API calls

### PR DESCRIPTION
**Pull request by matisidler**

### CODE REVIEW

File: internal/retry/retry.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	if maxElapsedTimeStr := os.Getenv("BACKOFF_MAX_ELAPSED_TIME"); maxElapsedTimeStr != "" {
+		if val, err := time.ParseDuration(maxElapsedTimeStr); err == nil && val > 0 {
+			params.MaxElapsedTime = val
+		}
+	}
+
+	logger.Stdout.Info("backoff params:",
+		slog.Duration("initial_interval", params.InitialInterval),
+		slog.Duration("max_interval", params.MaxInterval),
+		slog.Duration("max_elapsed_time", params.MaxElapsedTime),
```
Comment (matisidler) [inline internal/retry/retry.go]: these values aren't being printed for some reason, any idea? didn't have the chance to take a look at the logger's library

Comment (brody192) [reply internal/retry/retry.go]: They do for me, perhaps you didnt expand the context on the log line in Railway?

Comment (brody192) [reply internal/retry/retry.go:52]: They do for me, perhaps you didn't expand the context on the log line in Railway?

**Comment by brody192** [comment]
Hey, your retry logic was not retrying, I tested this by replacing the backboard URL with [https://utilities.up.railway.app/status-code/503](https://utilities.up.railway.app/status-code/503)

But I thank you for your initial implementation!

**Pull request merged**